### DISCUSSION TITLE
Issue while Deploying Karavan App using SB runtime

### DISCUSSION THREAD

**Discussion by ravishankar-hassain** [apache/camel-karavan#1056] [Q&A] [answered]
I am trying to deploy a sample camel Karavan application from vscode to minikube in my local setup. I am using the context menu Karavan:Deploy to do the deployment. The terminal output seems to be successful but the minikube pod is crashing with the below error

ravis@soleil_laptop MINGW64 ~
$ kubectl get pods
NAME READY STATUS RESTARTS AGE
karavan-poc-7875d74796-48xwf 0/1 CrashLoopBackOff 9 (2m45s ago) 24m

ravis@soleil_laptop MINGW64 ~
$ kubectl logs karavan-poc-7875d74796-48xwf
Starting the Java application using /opt/jboss/container/java/run/run-java.sh ...
INFO exec java -javaagent:/usr/share/java/jolokia-jvm-agent/jolokia-jvm.jar=config=/opt/jboss/container/jolokia/etc/jolokia.properties -javaagent:/usr/share/java/prometheus-jmx-exporter/jmx_prometheus_javaagent.jar=9779:/opt/jboss/container/prometheus/etc/jmx-exporter-config.yaml -XX:MinHeapFreeRatio=10 -XX:MaxHeapFreeRatio=20 -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -XX:+ExitOnOutOfMemoryError -cp ".:/deployments/*" org.springframework.boot.loader.JarLauncher
Error: Could not find or load main class org.springframework.boot.loader.JarLauncher
Caused by: java.lang.ClassNotFoundException: org.springframework.boot.loader.JarLauncher

I am using the latest version of Camel Karavan v4.3.0 extension in VSCode

I found the SO post https://stackoverflow.com/questions/77279480/org-springframework-boot-loader-jarlauncher-cannot-be-found-but-org-springframe

It seems like the JarLauncher class is actually under org.springframework.boot.loader.launch.JarLauncher for Springboot v3.2.0

but the Karavan / Camel JBang generated DockerFile has an ENV named JAVA_MAIN_CLASS with Value org.springframework.boot.loader.JarLauncher but for SB version 3.2 the value should be org.springframework.boot.loader.**launch**.JarLauncher

How can I change this? Is there an option to mention this via properties?

**Comment by mgubaidullin** [comment, marked as answer]
https://github.com/apache/camel-karavan/issues/1054
